app/generic/proof/proof.py

```diff
@@ -0,0 +1,217 @@
+import logging
+from enum import Enum
+from typing import Literal, Optional
+
+from aries_cloudcontroller import (
+    AcaPyClient,
+    IndyPresSpec,
+    IndyProofRequest,
+    V10PresentationSendRequestRequest,
+    V20PresRequestByFormat,
+    V20PresSendRequestRequest,
+    V20PresSpecByFormatRequest,
+)
+from fastapi import APIRouter, Depends
+
+from app.dependencies import agent_selector
+from app.generic.proof.facades.acapy_proof_v1 import ProofsV1
+from app.generic.proof.facades.acapy_proof_v2 import ProofsV2
+from app.generic.proof.models import PresentationExchange
+
+logger = logging.getLogger(__name__)
+
+
+router = APIRouter(prefix="/generic/proof", tags=["proof"])
+
+ProtocolVersion = Literal["1", "2"]
+
+
+class ProofsFacade(Enum):
+    v10 = ProofsV1
+    v20 = ProofsV2
+
+
+@router.post("/send-request")
+async def send_proof_request(
+    connection_id: str,
+    proof_request: IndyProofRequest,
+    protocol_version: Optional[ProtocolVersion] = "1",
+    aries_controller: AcaPyClient = Depends(agent_selector),
+) -> PresentationExchange:
+    """
+    Send proof request.
+
+    Parameters:
+    -----------
+    connection_id: str
+        The connection id
+    proof_request:
+        The proof request
+    protocol_version: Literal["1", "2"]
+        The protocol version. default is 1
+
+    Returns:
+    --------
+    presnetation_exchange: PresentationExchange
+        The presentation exchange record
+    """
+    try:
+        if protocol_version == "2":
+            presentation_exchange_rec = await ProofsFacade.v20.value.send_proof_request(
+                controller=aries_controller,
+                proof_request=V20PresSendRequestRequest(
+                    connection_id=connection_id,
+                    presentation_request=V20PresRequestByFormat(
+                        dif=None, indy=IndyProofRequest(**proof_request.dict())
+                    ),
+                ),
+                free=True,  # alway set this to TRUE because we only support this for now.
+            )
+            return presentation_exchange_rec
+        else:
+            presentation_exchange_rec = await ProofsFacade.v10.value.send_proof_request(
+                controller=aries_controller,
+                proof_request=V10PresentationSendRequestRequest(
+                    connection_id=connection_id,
+                    proof_request=IndyProofRequest(**proof_request.dict()),
+                ),
+                free=True,  # alway set this to TRUE because we only support this for now.
+            )
+        return presentation_exchange_rec
+    except Exception as e:
+        logger.error(f"Failed to create presentation record: \n{e!r}")
+        raise e from e
+
+
+@router.post("/create-request")
+async def create_proof_request(
+    proof_request: IndyProofRequest,
+    comment: Optional[str] = None,
```

Comment:
I'd combine as much parameters as possible in the body and nest them to be consistent with the issuer api

```
{
   "comment": "xxx",
   "proof_request": {...}
}